src/algebra/lie/non_unital_non_assoc_algebra.lean

```diff
@@ -23,9 +23,16 @@ algebra and we provide some basic definitions for doing so here.
 
 ## Main definitions
 
-  * `lie_ring.to_non_unital_non_assoc_semiring`
+  * `commutator_ring` turns a Lie ring into a `non_unital_non_assoc_semiring` by turning its
+    `has_bracket` (denoted `⁅, ⁆`) into a `has_mul` (denoted `*`).
   * `lie_hom.to_non_unital_alg_hom`
 
+## Implementation notes
+
+Previously the structure was added by a non-instance `lie_ring.to_non_unital_non_assoc_semiring`,
+which could be made an instance locally. This caused instance diamonds since all rings are
+also Lie rings, so we'd get two incompatible multiplications on each ring.
+
```

Comment:
I'd drop this: I don't think the history matters.
```suggestion
```
If you really want to mention the diamond, then I think a remark beside the instance would be better.

Reply:
A good place for this history is the PR description.